src/thor/timedep_forward.cc

```diff
@@ -259,14 +257,12 @@ TimeDepForward::GetBestPath(valhalla::Location& origin,
   Init(origin_new, destination_new);
   float mindist = astarheuristic_.GetDistance(origin_new);
 
-  // Set seconds from beginning of the week
-  seconds_of_week_ = DateTime::day_of_week(origin.date_time()) * midgard::kSecondsPerDay +
-                     DateTime::seconds_from_midnight(origin.date_time());
-
   // Initialize the origin and destination locations. Initialize the
   // destination first in case the origin edge includes a destination edge.
   uint32_t density = SetDestination(graphreader, destination);
-  SetOrigin(graphreader, origin, destination, seconds_of_week_);
+  // Call SetOrigin with kFreeFlowSecondOfDay for now since we don't yet have
+  // a timezone for converting a date_time of "current" to seconds_of_week
+  SetOrigin(graphreader, origin, destination, kFreeFlowSecondOfDay);
```

Comment:
I went with the free flow speed rather than `kInvalidSecondsOfWeek` since the latter resulted in failing astar tests for the trivial cases. Not sure why, still looking at that.

Reply:
Ok, the tests were failing beause they were either not setting the date-time on the origin/destination locations or setting it to 23:05 which pretty much matches `kFreeFlowSpeed` while a `kInvalidSecondsOfWeek` evaluates to `kConstrainedSpeed` which is speed around noon.

Setting the origin/destination datetimes to 13:05 rather than 23:05 and increasing the expected durations solved this so I can go back to `kInvalidSecondsOfWeek`.

Reply:
Pushed this here https://github.com/valhalla/valhalla/pull/2176/commits/9df751ec65fca263604289992b8fbda3603c5e56